targets/TARGET_STM/i2c_api.c

```diff
@@ -514,7 +525,7 @@ int i2c_byte_read(i2c_t *obj, int last) {
     timeout = FLAG_TIMEOUT;
     while (__HAL_I2C_GET_FLAG(handle, I2C_FLAG_RXNE) == RESET) {
         if ((timeout--) == 0) {
-            return -1;
+            return 2;
```

Comment:
that's how this is defined in the MBED HAL API of i2c_byte_read
https://github.com/ARMmbed/mbed-os/blob/master/hal/i2c_api.h#L144
> *  @return 0 if NAK was received, 1 if ACK was received, 2 for timeout.